internal/stolonctl/pkg/cluster/upgrade.go

```diff
@@ -39,6 +39,19 @@ const (
 )
 
 func Upgrade(ctx context.Context, config Config) error {
+	status, err := GetStatus(config)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	masterStatus, err := status.getMasterStatus()
+	if err != nil {
+		return trace.Wrap(err, "Cannot start upgrade.")
+	}
+	if !masterStatus.Healthy {
+		return trace.Errorf("Cannot start upgrade. Keeper master %s is unhealthy.",
```

Comment:
General consesus is to avoid the use of `trace.Errorf` and favor the typed error constructors instead.
It is not always easy to find the one that matches the intent and they're not extensible and might not map to the domain (well-known downsides). But I'd still use either `trace.BadParameter` or `trace.CompareFailed` instead of `trace.Errorf`.